views/js/runner/plugins/tools/textToSpeech/textToSpeech.js

```diff
@@ -0,0 +1,374 @@
+    'css!taoQtiTest/runner/plugins/tools/textToSpeech/textToSpeech',
+    'nouislider'
+], function (
+    $,
+    _,
+    __,
+    componentFactory,
+    tpl
+) {
+    'use strict';
+
+    /**
+     * The factory
+     * @param {jQuery} [options.$contentArea = '$(body)']
+     * @param {Boolean} [options.enableClickToSpeak = false]
+     * @param {Array} [options.ignoreEls = ['header', '.left-bar', '.right-bar', '.modal', 'footer', '.action-bar', '.sts-scope', '.media-container']]
+     * @param {String} [options.toolbarUrl = '//taotoolbar.speechstream.net/tao/configQA.js']
+     * @param {Object} [config]
+     * @returns {ui/component}
+     */
+    return function factory(options, config) {
+        var component;
+        var texthelpMapping = {
+            properties: {
+                pageCompleteCallback:    'eba_page_complete_callback',
+                texthelpSpeechStream:    'TexthelpSpeechStream',
+                volumeValue:             'eba_volume_value'
+            },
+            functions: {
+                barDynamicStart:         '$rw_barDynamicStart',
+                barCacheInit:            '$rw_barCacheInit',
+                barInit:                 '$rw_barInit',
+                cachePage:               '$rw_cachePage',
+                disableSpeech:           '$rw_disableSpeech',
+                enableClickToSpeak:      '$rw_enableClickToSpeak',
+                enableContinuousReading: '$rw_enableContinuousReading',
+                enableSpeech:            '$rw_enableSpeech',
+                event_pause:             '$rw_event_pause',
+                event_play:              '$rw_event_play',
+                getSpeed:                '$rw_getSpeed',
+                getVoice:                '$rw_getVoice',
+                hasReachedEnd:           '$rw_hasReachedEnd',
+                isPaused:                '$rw_isPaused',
+                isSpeaking:              '$rw_isSpeaking',
+                isTextSelectedForPlay:   '$rw_isTextSelectedForPlay',
+                isUsingMathjax:          '$rw_isUsingMathjax',
+                setSpeedValue:           '$rw_setSpeedValue',
+                setUsingMath:            '$rw_setUsingMath',
+                setUsingMathjax:         '$rw_setUsingMathjax',
+                setUsingMaths:           '$rw_setUsingMaths',
+                setVolumeValue:          '$rw_setVolumeValue',
+                stopSpeech:              '$rw_stopSpeech',
+                tagSentences:            '$rw_tagSentences',
+                userParameters:          '$rw_userParameters'
+            }
+        };
+        var speed;
+        var volume;
+
+        _.assign(options || {}, {
+            contentArea: $('body'),
+            enableClickToSpeak: false,
+            ignoreEls: ['header', '.left-bar', '.right-bar', '.modal', 'footer', '.action-bar', '.sts-scope', '.media-container'],
+            toolbarUrl: '//taotoolbar.speechstream.net/tao/configQA.js'
```

Comment:
not familiar with this API but why does the config need to be loaded remotely ? Could we not host our own version of it ?

Reply:
There are multiple versions of this file and thus needs to be a configurable option. 

I will say there is potential to host this particular file ourselves, as the different versions I believe differ only on configuration options set within that file. Meaning, we could update the config options ourselves and host this file.

An argument against would be that we remove Texthelp's access to this file. In which case, we would need to reupload any updates they might wish to add to these config files.

My personal thought is that Texthelp is a service we consume and that we should push as much work to their end (including config file hosting) as much as possible. That is, keep as much of Texthelp off of tao as we can.